src/compose/ComposeBox.js

```diff
@@ -403,12 +409,16 @@ class ComposeBox extends PureComponent<Props, State> {
       isAdmin,
       isAnnouncementOnly,
       isSubscribed,
-      jitsiServerUrl,
       stream,
+      videoChatProvider,
     } = this.props;
 
+    // Flow is erroneously widening the type of videoChatProvider to also allow
+    // `string` types here. The type is refined back down to the correct type by
+    // explicitly annotating `ComposeBox.prototype.insertVideoCallLink`, but be
+    // aware of that if you need to use this value outside of that method.
```

Comment:
Hmm, what's the type that you're seeing widened to `string`? When I remove the annotation on the argument type of `insertVideoCallLink`, and then hover on various tokens here (in VS Code) to see the types Flow infers, I don't spot any `string` types.

Over in the implementation of `insertVideoCallLink`, if I hover on the `name` in `videoChatProvider.name`, I do see `string | "jitsi_meet"`. That is puzzling. But it seems to be the same with or without that type annotation on the `videoChatProvider` argument. Fortunately it also doesn't seem to do any harm.

Reply:
Huh, that's puzzling indeed. I'm updating my comment to clarify that it's `videoChatProvider.name` that's being widened. One occasion where this would cause problems is if you were to do something like this:

```javascript
let isBigBlueButton = false;
if (videoChatProvider !== null) {
  // For the purposes of this example, the correct name for the below provider is `big_blue_button`.
  if (videoChatProvider.name === 'bigbluebutton') {
    isBigBlueButton = true;
  }
}
```

The above example passes flow's checker when done in the body of the render method, but strangely, it doesn't pass if moved into the body of `insertVideoCallLink`, despite the fact that the type widening appears to be happening there too, as you correctly pointed out.